web-search-engines/langchain4j-web-search-engine-searchapi/src/main/java/dev/langchain4j/web/search/searchapi/SearchApiRequest.java

```diff
@@ -0,0 +1,105 @@
+package dev.langchain4j.web.search.searchapi;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static dev.langchain4j.internal.Utils.getOrDefault;
+import static dev.langchain4j.internal.Utils.isNotNullOrBlank;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+@Builder
+class SearchApiRequest {
+
+	/************************************************************************************************ 
+	 * Mandatory parameters 
+	 * **********************************************************************************************/
```

Comment:
does this internal class really needed to be documented so heavily? this class won't be used by LangChain4J clients. If yes, then pls all these comments shall be reformatted, see https://github.com/langchain4j/langchain4j/blob/d28f5ab470360f68a688bbe0a2ed952d07ef9bf9/langchain4j-core/src/main/java/dev/langchain4j/web/search/WebSearchRequest.java as example

Reply:
I re-did the formatting of the comment.

Reply:
The clean javadoc comment in `WebSearchRequest` starting the line 11. And see how nice and readable is the generated documentation as result: 
![image](https://github.com/langchain4j/langchain4j/assets/33568148/980e186c-6582-48b9-ba06-316cec80e0ee)

Reply:
Agreed 👍.

I've redone the comments to follow the javadoc style as requested.